compiler/rustc_ast_lowering/src/format.rs

```diff
@@ -12,6 +12,34 @@ use rustc_span::{
 };
 use std::borrow::Cow;
 
+fn int_ty_max(int_ty: IntTy) -> Option<u128> {
+    match int_ty {
+        // isize is platform-dependent, so we should use
+        // TyCtxt.data_layout.pointer_size
+        // This is available, for example, from a LoweringContext
+        IntTy::Isize => None,
```

Comment:
I think it would be better to pass in the target's `isize_max` as an argument to this function (and similarly with `usize_max` for the `uint_ty_max` function below) and return the max as a plain `u128` rather than wrapped in an `Option` (the semantics of which are not obvious).

Reply:
Makes sense, but it looked a bit weird to me to pass another argument to these functions. So I changed these to methods on `LoweringContext` instead, so that the *size_max computation can be done in these functions instead.